packages/SwingSet/src/kernel/state/storageHelper.js

```diff
@@ -2,13 +2,26 @@
 
 import { assert } from '@agoric/assert';
 
+export function* enumeratePrefixedKeys(kvStore, prefix) {
+  // return an iterator of all existing keys that start with
+  // ${prefix}, in lexicographic order, excluding ${prefix} itself
+  let key = prefix;
+  for (;;) {
+    key = kvStore.getNext(key);
```

Comment:
As I've been reading through this, I'm increasingly thinking the `getNext` function should be named `getNextKey` or maybe `getKeyAfter` -- the main idea being that the result is a key string and not a value string, in contrast to `get` which returns a value.

Reply:
Also, I'm thinking that the notional `getNextEntry` call you outlined above may be something we'll want sooner rather than later, not so much for performance/efficiency but for expressiveness.

Reply:
BTW the name `getNext` comes from a story I was told about the early days of the SNMP protocol, with which routers and switches could publish information like how many packets have been sent through a certain ethernet port. The design has (had? this was a long long time ago) only two API methods, and they were described to me as "`get`, and the mighty `getNext`". I had the impression that the author of the protocol introduced the Mighty qualifier, but I haven't been able to find a reference to that, so maybe it was just the coworker who introduced me to it. But the idea was that these two methods were all you needed to enumerate the whole tree of queryable properties, and `getNext` was the interestingly-powerful one.

(that said, I don't actually know whether the SNMP `getNext` returned just a key, or a key/value entry pair.. I should dig that up if I'm going to follow their precedent)

Anyways, yeah there's also a tradeoff of how short we can keep the method names, vs the ambiguity we might introduce.